marc_to_solr/lib/action_note_builder.rb

```diff
@@ -0,0 +1,91 @@
+class ActionNoteBuilder
+  attr_reader :record
+
+  def initialize(record:)
+    @record = record
+  end
+
+  def self.build(record:)
+    new(record:).build
+  end
+
+  def build
+    notes = []
+
+    @record.fields('583').each do |field|
+      public_note = field.indicator1 == "1"
+      field_link = field["8"]
+      next unless public_note && (field_link.present? || pulfa_record? || scsb_record?)
+
+      notes << { description: description(field), uri: uri(field) }
+    end
+    notes = [notes.to_json.to_s] if notes.present?
+    notes.presence
+  end
+
+  private
+
+    def description(field)
+      description = ""
+      description << materials_specified(field)
+      description << action(field)
+      description << action_interval(field)
+      description << authorization_phrase(field)
+      description << institution(field)
+    end
+
+    def uri(field)
+      return '' if field["u"].blank?
+
+      field["u"]
```

Comment:
Is there a chance there could be leading or trailing whitespace that needs to be removed from this subfield value?

Reply:
I don't *think* so, but checking in with Christina and Mark to double check.

Reply:
I added a `.strip`, since Mark said Alma doesn't strip them.